heroic-component/src/main/java/com/spotify/heroic/tracing/EnvironmentMetadata.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2020 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.heroic.tracing;
+
+import static com.spotify.heroic.tracing.EnvVariableLookup.getEnvVariable;
+import static io.opencensus.trace.AttributeValue.stringAttributeValue;
+import static java.util.Collections.unmodifiableMap;
+import static java.util.Objects.isNull;
+
+import io.opencensus.trace.AttributeValue;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class EnvironmentMetadata implements TracingMetadataProvider {
+    /** The absolute max size of attributes, including optional ones */
+    private static final int MAX_ATTRIBUTE_SIZE = 8;
+    private static Map<String, AttributeValue> attributes;
+
+    private EnvironmentMetadata() { }
+
+    private static void addGcpMetadata() {
+        String projectId = GcpMetadataLookup.getProjectId();
+
+        if (isNull(projectId) || projectId.isBlank()) {
+            return;
+        }
+
+        attributes.put("gcp.project", stringAttributeValue(GcpMetadataLookup.getProjectId()));
+        String zone = GcpMetadataLookup.getZone();
+        attributes.put("gcp.zone", stringAttributeValue(zone));
+        attributes.put(
+            "gcp.region", stringAttributeValue(zone.substring(0, zone.lastIndexOf("-"))));
+    }
+
+    private static void addHostname() {
+        final Optional<String> hostname = getEnvVariable("HOSTNAME");
```

Comment:
I think it makes more sense to have this as part of the tracing config so everything can continue to be set in the heroic config file.

That might require using dagger instead of directly creating a singleton. I haven't fully thought through the instantiation but it feels more in-line with the rest of Heroic/tracing config loading.

Reply:
I like the idea of just moving this to the tracing config as a list of tags, and setting the hostname tag could be done through config management. I do think it needs to be configurable - neither the env var HOSTNAME nor `gethostname` calls necessarily return a meaningful global identifier, and having to configure those just for the purpose of this tag doesn't seem great.

Example: on my local system, HOSTNAME is not set and `gethostname` returns a single non-domained host. In k8s, HOSTNAME is set but both it and `gethostname` are just the pod name without namespace/cluster/region/etc.

Reply:
you can now set tags via the config.